tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/arena/ElemPtr.scala

```diff
@@ -0,0 +1,65 @@
+package at.forsyte.apalache.tla.bmcmt.arena
+
+import at.forsyte.apalache.tla.bmcmt.ArenaCell
+import at.forsyte.apalache.tla.lir.TypedPredefs.BuilderExAsTyped
+import at.forsyte.apalache.tla.lir.convenience.tla
+import at.forsyte.apalache.tla.lir.{BoolT1, TlaEx, UID}
+
+/**
+ * An abstract membership pointer.
+ */
+sealed trait ElemPtr {
+
+  /**
+   * Get the element that this edge is pointing to.
+   * @return
+   *   an arena cell that is pointed by this pointer
+   */
+  def elem: ArenaCell
+
+  /**
+   * Translate the membership test into an expression that can be understood by Z3SolverContext.
+   */
+  def toSmt: TlaEx
+}
+
+/**
+ * An element pointer that always evaluates to a fixed Boolean value.
```

Comment:
I think all of these case classes could use an example of when to use them, as given in the ADR.